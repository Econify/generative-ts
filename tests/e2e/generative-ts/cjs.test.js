const cjs = require("../../../packages/generative-ts/dist/index.cjs");

describe('generative-ts cjs bundle', () => {
  test('api snapshot', async () => {
    expect(cjs).toMatchSnapshot();
  });

  test('sanity check', async () => {
    const model = cjs.createLmStudioModelProvider({
      modelId: "TheBloke/Mistral-7B-Instruct-v0.2-GGUF",
    });
    
    const response = await model.sendRequest({
      $prompt: "Say hi:",
      generation_config: {
        max_output_tokens: 50,
        temperature: 0
      }
    });

    expect(response.choices[0].message).toBeDefined();
  });
});
