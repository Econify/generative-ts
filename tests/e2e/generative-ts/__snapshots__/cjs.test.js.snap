// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generative-ts cjs bundle api snapshot 1`] = `
{
  "Ai21Jurassic2Api": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Ai21Jurassic2Template": i {
    "source": [Function],
  },
  "AmazonTitanTextApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "AmazonTitanTextTemplate": i {
    "source": [Function],
  },
  "BaseModelProvider": [Function],
  "CohereChatApi": {
    "name": "COHERE-CHAT",
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "CohereChatTemplate": i {
    "source": [Function],
  },
  "CohereGenerateApi": {
    "name": "COHERE-GENERATE",
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "CohereGenerateTemplate": i {
    "source": [Function],
  },
  "FnTemplate": [Function],
  "GoogleGeminiApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "GoogleGeminiTemplate": i {
    "source": [Function],
  },
  "HfConversationalTaskApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "HfConversationalTaskTemplate": i {
    "source": [Function],
  },
  "HfTextGenerationTaskApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "HfTextGenerationTaskTemplate": i {
    "source": [Function],
  },
  "HttpModelProvider": [Function],
  "Llama2ChatApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Llama2ChatTemplate": i {
    "source": [Function],
  },
  "Llama3ChatApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Llama3ChatTemplate": i {
    "source": [Function],
  },
  "MistralAiApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "MistralAiTemplate": i {
    "source": [Function],
  },
  "MistralBedrockApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "MistralBedrockTemplate": i {
    "source": [Function],
  },
  "OpenAiChatApi": {
    "requestTemplate": i {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "OpenAiChatTemplate": i {
    "source": [Function],
  },
  "createAwsBedrockModelProvider": [Function],
  "createCohereModelProvider": [Function],
  "createGroqModelProvider": [Function],
  "createHuggingfaceInferenceModelProvider": [Function],
  "createLmStudioModelProvider": [Function],
  "createMistralModelProvider": [Function],
  "createOpenAiChatModelProvider": [Function],
  "createVertexAiModelProvider": [Function],
}
`;
